World: handle name collisions like Model

scpeters opened the pull request:
# 🦟 Bug fix

Follow-up to https://github.com/gazebosim/sdformat/pull/1293#discussion_r1287959178

## Summary



## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

scpeters commented on src/World.cc:
```diff
@@ -283,21 +285,35 @@ Errors World::Load(sdf::ElementPtr _sdf, const ParserConfig &_config)
+        ss << "Frame with name [" << frame.Name() << "] "
+           << "in world with name [" << this->Name() << "] "
+           << "has a name collision, changing frame name to ["
+           << frameName << "].";
```
related to https://github.com/gazebosim/sdformat/issues/1312

I can revert this if desired

scpeters commented:
this is a behavior change, so we could consider targeting `main`

azeey commented:
> this is a behavior change, so we could consider targeting `main`

It's probably best if we target `main`

scpeters commented:
> > this is a behavior change, so we could consider targeting `main`
> 
> It's probably best if we target `main`

I rebased on `main` and updated the Migration guide in https://github.com/gazebosim/sdformat/pull/1311/commits/75bb46533cc4f88c40d4278d7168f9856222ae5c
